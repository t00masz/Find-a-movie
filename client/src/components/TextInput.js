import React from 'react';
import changeUserData from '../actions/user-actions';
import { connect } from "react-redux";

const TextInput = (props) => {
  return (
  <div className="option">
    <input className="option__input" name={props.name} type='text' placeholder={props.placeholder} value={props.value} onChange = {(e) => {
      props.dispatch(changeUserData(props.name, e.target.value))
    }}/>    
  </div>
  )
};

export default connect()(TextInput);
