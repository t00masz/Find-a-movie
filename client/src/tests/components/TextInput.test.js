import React from 'react';
import { shallow } from 'enzyme'
import TextInput from '../../components/SubmitInput';
import user from '../fixtures/user';

test('should render text input', () => {
    const wrapper = shallow(<TextInput 
        name="name" 
        placeholder="Your first name (required)" 
        type="text"
        pattern="([a-zA-Z]){2,30}"
        value={user.name} />);
    expect(wrapper).toMatchSnapshot();
});
