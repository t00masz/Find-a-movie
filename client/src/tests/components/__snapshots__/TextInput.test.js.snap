// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitInput
    name="name"
    pattern="([a-zA-Z]){2,30}"
    placeholder="Your first name (required)"
    type="text"
    value="Jan"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="button"
        type="submit"
        value={undefined}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "button",
        "type": "submit",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="button"
          type="submit"
          value={undefined}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "button",
          "type": "submit",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
